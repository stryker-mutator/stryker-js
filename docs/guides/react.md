---
title: React
custom_edit_url: https://github.com/stryker-mutator/stryker-js/edit/master/docs/guides/react.md
---

Stryker supports React projects using Jest with both JSX and TSX code.

## JSX project

Recommended stryker packages: `npm i -D @stryker-mutator/core @stryker-mutator/jest-runner`

Recommended other packages:

- jest 23.0.0 or higher

### Configuration

After installing the recommended packages, create the `stryker.conf.json` file in your repository.
The configuration below contains a good starting point for React projects.
You may have to change some paths like the [mutate](../configuration.md#mutate-string) array.

```json
{
  "testRunner": "jest",
  "jest": {
    "projectType": "create-react-app"
  }
}
```

## TSX projects

For projects using TypeScript and TSX, you can follow the JSX guide but with a few differences:

Recommended stryker packages: `npm i -D @stryker-mutator/core @stryker-mutator/jest-runner @stryker-mutator/typescript-checker`

Configuration:

```json
{
  "testRunner": "jest",
  "jest": {
    "projectType": "create-react-app"
  },
  "checkers": ["typescript"],
  "tsconfigFile": "tsconfig.json"
}
```

## Troubleshooting

Please take a look at the [troubleshooting page](../troubleshooting.md) when you run into any problems setting up StrykerJS.
