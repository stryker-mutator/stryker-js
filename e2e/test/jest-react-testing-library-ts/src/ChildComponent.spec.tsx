import React from 'react';
import { render, screen } from '@testing-library/react';
import ChildComponent from './ChildComponent';

describe('Child Component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ChildComponent text="hello" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
