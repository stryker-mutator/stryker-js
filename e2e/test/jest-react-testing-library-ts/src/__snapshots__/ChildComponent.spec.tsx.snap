// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="test"
  >
    hello
  </div>
</DocumentFragment>
`;
